core/src/main/java/io/hyperfoil/core/steps/AwaitDelayStep.java

```diff
@@ -24,7 +24,7 @@ public AwaitDelayStep(ReadAccess key) {
    @Override
    public boolean invoke(Session session) {
       ScheduleDelayStep.Timestamp blockedUntil = (ScheduleDelayStep.Timestamp) key.getObject(session);
-      return System.currentTimeMillis() >= blockedUntil.timestamp;
+      return System.nanoTime() >= blockedUntil.timestamp;
```

Comment:
compute the distance here - since System.nanoTime can wrap (negative to positive) - the diff is always safe due to math oveflow - see doc at https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#nanoTime--

Reply:
Thanks for the suggestion! Updated and added a comment as well